crates/transparency/src/map/map.rs

```diff
@@ -224,51 +185,18 @@ where
         let mut here = self.clone();
 
         for (key, val) in iter {
-            let mut path = Path::new(&key);
-            let leaf = hash_leaf(key, val);
-            let (node, new) = here.link.node().insert(&mut path, leaf);
+            let mut path: Path<D, K> = Path::new(key.clone());
+            let (node, new) = here.link.node().insert(&mut path, key, Hash::of(val));
             here = Self::new(Link::new(node), here.len + usize::from(new));
         }
 
         here
     }
 }
 
-/// Hashes a leaf node
-/// ```hash
-/// H(0xff || H(<key>) || <value>)
-/// ```
-pub(crate) fn hash_leaf<D, K, V>(key: K, value: V) -> Hash<D>
-where
-    D: SupportedDigest,
-    K: VisitBytes,
-    V: VisitBytes,
-{
-    let key_hash: Hash<D> = Hash::of(&key);
-    Hash::of(&(0xffu8, key_hash, value))
-}
-
-/// ```hash
-/// // Both children present:
-/// H(0b11 || <left> || <right>)
-///
-/// // Left child present:
-/// H(0b10 || <left>)
-///
-/// // Right child present:
-/// H(0b01 || <right>)
-///
-/// // Empty tree:
-/// H(0b00)
-/// ```
-pub(crate) fn hash_branch<D>(lhs: Option<Hash<D>>, rhs: Option<Hash<D>>) -> Hash<D>
+pub(crate) fn hash_branch<D>(lhs: Hash<D>, rhs: Hash<D>) -> Hash<D>
```

Comment:
Can we pass `lhs` and `rhs` by reference? Doesn't seem like this needs to be by-value.